npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb

```diff
@@ -365,11 +365,12 @@ def dependencies_in_error_message?(error_message)
         def write_temporary_dependency_files(lockfile_name,
                                              update_package_json: true)
           write_lockfiles(lockfile_name)
-          File.write(".npmrc", npmrc_content)
 
           package_files.each do |file|
             path = file.name
-            FileUtils.mkdir_p(Pathname.new(path).dirname)
+            working_directory = Pathname.new(path).dirname
+            FileUtils.mkdir_p(working_directory)
+            File.write("#{working_directory}/.npmrc", npmrc_content)
```

Comment:
Should be able to ✂️ this if the above suggestion works
```suggestion
```

Reply:
@feelepxyz I think I tried your suggestion as my first iteration but I seem to recall it didn't quite work, but I might be wrong, I'll give it another shot because I do also prefer to avoid writing files all the time within the loop.